clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -960,7 +960,7 @@ void maybeReconcile() {
         // best effort to commit the offsets in the case where the epoch might have changed while
         // the current reconciliation is in process. Note this is using the rebalance timeout as
         // it is the limit enforced by the broker to complete the reconciliation process.
-        commitResult = commitRequestManager.maybeAutoCommitSyncBeforeRevocation(getExpirationTimeForTimeout(rebalanceTimeoutMs));
+        commitResult = commitRequestManager.maybeAutoCommitSyncBeforeRevocation(getDeadlineMsForTimeout(rebalanceTimeoutMs));
```

Comment:
I might misunderstand your question—are you suggesting to pass in the _timeout_ vs. the _deadline_? There's no `Timer` in the `getDeadlineMsForTimeout`. It just calculates the deadline from the timeout and handles overflow.

Reply:
My assumption was that the whole reconciliation is limited by the rebalance timeout. But here it seems that we only have a timeout on the auto commit sync. I would have expected that different parts of the reconciliation would update a timer and the commit sync would take the remaining time as the timeout.
But maybe I missing something.

Reply:
Hey @cadonna, your assumption is right: "the whole reconciliation is limited by the rebalance timeout", but the trick is that it's the broker who enforces that, not the client.

That's why we don't see the `rebalanceTimeout` being used on the client side, other than for sending it to the broker in the initial HB, and here to limit the commit retries. We use it on the commit only because we needed to put a limit on the retries to make sure that if the commit keeps failing and being retried, the request is not kept forever on the client after the broker kicks the member out of the group.

Reply:
Ah, I see! That makes sense! Then I propose to rename the field to something like `commitTimeoutDuringReconciliation` to make that clear.

Reply:
Created KAFKA-16899 as a little bit of thought needs to be given to how the name change is handled.

Reply:
You can do it as a MINOR pull request, if you want, but it really is just a renaming of that variable and of the parameter in the constructor as you correctly stated in the ticket. Regarding your point in the ticket about there is more to it than changing the name, I cannot completely follow. When somebody instantiates a `MembershipManagerImpl` and passes the config `rebalance.timeout.ms` into the constructor for parameter `commitTimeoutDuringReconciliation` they state that they want to use the value of config `rebalance.timeout.ms` as `commitTimeoutDuringReconciliation`. This seems quite clear to me.